examples/jsm/loaders/PCDLoader.js

```diff
@@ -410,6 +425,22 @@ class PCDLoader extends Loader {
 
 				}
 
+				if ( offset.red !== undefined ) {
+
+					const red = dataview.getFloat64( row + offset.red , true);
+					const green = dataview.getFloat64( row + offset.green , true);
+					const blue = dataview.getFloat64( row + offset.blue , true);
```

Comment:
The problem is your code is application specific since it does not base on a specification. You are assuming here that R,G,B fields are floats but what if other tools report `uint8` values? 

Like mentioned above, such modifications are not reliable in library code. `PCDLoader` isn't a large module so it's best to copy the module and add custom properties on application level.

Reply:
You are right, I can change this. But you are willing to add this capability in general aren't you?. Your first response read more like a hard no. Hence, I am asking.

Reply:
Um, I wonder if we eventually need a more generic approach in `PCDLoader` similar to what #27901 has suggested.

The loader could parse the existing default fields like `x`, `y`, `z` and return a geometry as before. But now the loader also parses all other fields it finds in the PCD asset. The related data could be returned as a simple map (name > typed array) in the `userData` field of the geometry. Ideally the geometry only hold fields in attributes that can be actually rendered.